app/environments/brassbirmingham/brassbirmingham/envs/brassbirmingham.py

```diff
@@ -0,0 +1,384 @@
+
+import gym
+import numpy as np
+import random
+
+import config
+
+from stable_baselines import logger
+
+from .classes import *
+
+class BrassBirminghamEnv(gym.Env):
+    metadata = {'render.modes': ['human']}
+
+    def __init__(self, verbose = False, manual = False):
+        super(BrassBirminghamEnv, self).__init__()
+        self.name = 'brassbirmingham'
+        self.n_players = 3
+
+        self.manual = manual
+
+        self.board_size = 7
+        self.squares = self.board_size * self.board_size
+
+        self.tile_types = 11
+        
+        self.max_score = 100
+        
+        self.total_positions = self.squares + self.n_players + 1
+
+        self.set_contents()
+        self.nets = [5,7,16, 24, 32, 41, 43]
+        self.total_tiles = sum([x['count'] for x in self.contents])
+
+        self.action_space = gym.spaces.Discrete(self.total_tiles  * 2)
+        self.observation_space = gym.spaces.Box(0, 1, (self.total_positions * self.total_tiles + self.squares + 4 + self.n_players + self.action_space.n ,))
+        self.verbose = verbose
+
+
+    def set_contents(self):
+        self.contents = []
+
+        for colour in ['R','B','G','Y']:
+            for value in range(1, 6):
+                self.contents.append({'tile': BrassBirmingham, 'info': {'name': f'{colour}{value}brassbirmingham', 'colour': colour, 'value': value}, 'count': 2} )
+            self.contents.append({'tile': BrassBirmingham, 'info': {'name': f'{colour}brassbirmingham', 'colour': colour, 'value': 0}, 'count': 1} )
+
+        self.contents.append({'tile': Flower, 'info': {'name': 'flower'}, 'count':  13})
+
+        for value in range(1,10):
+            self.contents.append({'tile': Dragonfly, 'info': {'name': 'dragonfly', 'value': value}, 'count':  1})
+
+        for value in range(1,10):
+            self.contents.append({'tile': LightningBug,  'info': {'name': 'lightningbug', 'value': value}, 'count':  1})
+
+        for value in range(1,10):
+            self.contents.append({'tile': Cricket, 'info': {'name': 'cricket', 'value': value}, 'count':  1})
+
+        self.contents.append({'tile': Bee, 'info': {'name': 'bee'}, 'count':  6})
+
+        for value in range(10,16):
+            self.contents.append({'tile': Honeycomb, 'info': {'name': 'honeycomb', 'value': value}, 'count':  1})
+        
+        for value in range(-4,-8, -1):
+            self.contents.append({'tile': Wasp, 'info': {'name': 'wasp', 'value': value}, 'count':  1})
+
+        
+    @property
+    def observation(self):
+        obs = np.zeros(([self.total_positions, self.total_tiles]))
```

Comment:
To prevent confusion, could we just remove all this stuff that isn't part of BB and add a # TODO and maybe point to the line in the other file we could use for example?

Reply:
sounds good